JustSaying/Extensions/TypeExtensions.cs

```diff
@@ -15,18 +14,7 @@ public static string ToTopicName(this Type type)
                 ? Regex.Replace(type.FullName, "\\W", "_").ToLower()
                 : type.Name.ToLower();
 
-            if (name.Length > MAX_TOPIC_NAME_LENGTH)
-            {
-                var suffix = name.GetInvariantHashCode().ToString();
-                name = name.Substring(0, MAX_TOPIC_NAME_LENGTH - suffix.Length) + suffix;
-            }
-
-            return name;
-        }
-
-        private static int GetInvariantHashCode(this string value)
-        {
-            return value.Aggregate(5381, (current, character) => (current*397) ^ character);
+            return name.TruncateTo(MAX_TOPIC_NAME_LENGTH);
```

Comment:
There should be tests on this

Reply:
Have added some (though I haven't added any new functionality here btw - just moved the truncation code elsewhere so it can be reused)